include/lsst/afw/cameraGeom/TransformMap.h

```diff
@@ -0,0 +1,213 @@
+     *         `fromCoordSys` or `toCoordSys` is not supported.
+     */
+    geom::Point2D transform(geom::Point2D const &point, CameraSys const &fromCoordSys,
+                            CameraSys const &toCoordSys) const;
+
+    /**
+     * Convert a list of points from one coordinate system to another.
+     *
+     * @overload
+     */
+    std::vector<geom::Point2D> transform(std::vector<geom::Point2D> const &pointList,
+                                         CameraSys const &fromCoordSys, CameraSys const &toCoordSys) const;
+
+    CameraSysIterator begin() const { return boost::make_transform_iterator(_frameIds.begin(), GetKey()); }
+
+    CameraSysIterator end() const { return boost::make_transform_iterator(_frameIds.end(), GetKey()); }
+
+    /**
+     * Can this transform to and from the specified coordinate system?
+     *
+     * @param system the coordinate system to search for
+     * @returns `true` if `system` is supported, `false` otherwise
+     *
+     * @exceptsafe Shall not throw exceptions.
+     */
+    bool contains(CameraSys const &system) const noexcept;
+
+    /**
+     * Get a Transform from one coordinate system to another.
+     *
+     * @param fromCoordSys, toCoordSys coordinate systems between which to transform
+     * @returns an invertible Transform converting from `fromCoordSys` to `toCoordSys`.
+     *
+     * @throws lsst::pex::exceptions::InvalidParameterError Thrown if either
+     *         `fromCoordSys` or `toCoordSys` is not supported.
+     */
+    std::shared_ptr<Transform> getTransform(CameraSys const &fromCoordSys, CameraSys const &toCoordSys) const;
+
+    /**
+     * Get the number of supported coordinate systems.
+     *
+     * @exceptsafe Shall not throw exceptions.
+     */
+    size_t size() const noexcept;
+
+private:
+    /**
+     * The internal frame ID corresponding to a coordinate system.
+     *
+     * @param system The system to convert.
+     * @return the ID by which the coordinate system can be found in transforms.
+     *
+     * @throws lsst::pex::exceptions::InvalidParameterError Thrown if `system`
+     *         is not in frameIds.
+     *
+     * @exceptsafe Provides strong exception guarantee.
+     */
+    int _getFrame(CameraSys const &system) const;
+
+    /**
+     * An ast::Mapping that transforms between two coordinate systems.
+     *
+     * @param fromCoordSys, toCoordSys coordinate systems between which to transform
+     * @return an invertible Mapping that converts from `fromCoordSys` to `toCoordSys`
+     *
+     * @throws lsst::pex::exceptions::InvalidParameterError Thrown if either
+     *         `fromCoordSys` or `toCoordSys` is not supported.
+     */
+    std::shared_ptr<ast::Mapping const> _getMapping(CameraSys const &fromCoordSys,
+                                                    CameraSys const &toCoordSys) const;
+
+    /// Allows conversions between LSST and AST data formats
+    static geom::Point2Endpoint _pointConverter;
```

Comment:
Could this be in an anonymous namespace in the .cc file instead?  I suppose it's a bit of a judgement call which is better, but I like minimizing what goes into header files.